src/packages/utils/src/utils/request-coordinator.ts

```diff
@@ -74,6 +77,23 @@ export class RequestCoordinator {
     }
   };
 
+  public disconnect() {
+    this.pause();
+    // make this async to force a Promise return type
+    this.queue = async () => {
+      throw new Error("Cannot process request, Ganache is disconnected.");
+    };
+    // ensure that processing cannot be resumed
+    this.resume = () => {
+      throw new Error("Cannot resume processing requests, Ganache is disconnected.");
+    }
+
+    while (this.pending.length > 0) {
+      const current = this.pending.shift();
```

Comment:
If the order of this flush doesn't matter, does it make sense to use `pop` instead of `shift` in this case? `pop` updates a property (length likely), `shift` would have to shift every remaining element by 1.

Reply:
I thought about that - and it _probably_ doesn't matter, but I figured respecting the FIFO nature of the queue makes sense. 

It may be marginally faster, but this is only happening during disconnection _if_ there's anything queued up, so I don't imagine it'll have much of an impact 🤷

Reply:
I added a comment explaining

Reply:
Hm.... I don't think we need to purge pending tasks since we can't actually stop them anyway, i'm pretty sure `blockchain.stop()` waits until all pending operations are complete before resolving, so forcing the requests to be `reject`ed early in the RequestCoordinator doesn't shut us down any faster.

Reply:
We're only rejecting pending tasks, which haven't been executed yet.

Once they get to the head of the queue, they're `shift`ed from `pending` before execution, meaning they're unavailable to the purging function.

Reply:
Ah, I was mistaken!

Reply:
Mowing the lawn again, thinking about code, per usual, and I had a realization: these shouldn't be rejected until after Ganache in flight requests have been fulfilled. Ganache let's users opt out of asyncRequestProcessing which means requests will be processed in the order they are received. I think this should hold true even when Ganache is shutting down.

So I think we need to pause the coordinator, shut down everything else (which already waits for in-flight requests to be fulfilled), and _then_ reject these. Feel free to try to change my mind though!

Reply:
Ah, that's unfortunate. In that case I think this PR should only do what it set out to do.